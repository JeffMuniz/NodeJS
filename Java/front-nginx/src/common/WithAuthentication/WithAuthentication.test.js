import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import PrivateRoute from "./WithAuthentication";

jest.mock("../../scenes/Login/Login.container.js", () => <div>Login</div>);
jest.mock("react-router-dom");
jest.mock("@common", () => ({ Template: "Template" }));

describe("WithAuthentication - Unit Test", () => {
  it("Should render component with default props", () => {
    // given
    const renderer = new ShallowRenderer();
    const props = {
      exact: true,
      path: "/fake",
      computedMatch: {
        from: "/lorem",
        url: "/lorem",
      },
      component: () => <div>Home</div>,
    };

    // when
    renderer.render(<PrivateRoute {...props} />);
    const result = renderer.getRenderOutput();
    // then
    expect(result).toMatchSnapshot();
  });
});
