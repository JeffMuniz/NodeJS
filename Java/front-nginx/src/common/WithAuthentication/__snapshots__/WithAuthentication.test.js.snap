// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithAuthentication - Unit Test Should render component with default props 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/login",
      "state": null,
    }
  }
/>
`;
